### PR TITLE
Server role allocation

### PR DESCRIPTION
**Pull request by jessetilro**
The role allocation is now verified server-side. The process of joining the game will be further tuned in upcoming PR's, fixing networking issues.

Was developed and should therefore be tested in parallel with PR https://github.com/jessetilro/pixelperfect/pull/319.

### CODE REVIEW

**Comment by jessetilro** [comment]
Working on fixing the build, please bear with me.

File: app/src/main/java/nl/tudelft/pixelperfect/client/ClientListener.java [hunk 1]
```diff
@@ -28,9 +29,29 @@
...
+        System.out.println("Received role chosen message.");
+        if (message.isAllocated()) {
+            System.out.println("Server granted request for " + message.getRole().toString());
+            RoleActivity.enterLobby(message.getRole());
+        } else {
+            //RoleActivity.updateRoleAvailability((RoleChosenMessage) message);
+            //StringBuilder sb = new StringBuilder();
+            //sb.append("The role ").append(chosenRole.getRole().toString()).append(" is already taken!");
+            //RoleActivity.showMessage(sb.toString());
         }
```
Comment (FlXlT) [inline app/src/main/java/nl/tudelft/pixelperfect/client/ClientListener.java:55]: Commented-out code.

**Comment by wouterz** [comment]
Looks good, doing some manual testing now

File: app/src/main/java/nl/tudelft/pixelperfect/client/ClientListener.java (hunk 1 above)
Comment (jessetilro) [inline app/src/main/java/nl/tudelft/pixelperfect/client/ClientListener.java:55]: I knew you would notice this, I did this on purpose. Will either fix or remove it next PR.

**Comment by wouterz** [comment]
[copied from [server PR](jessetilro/pixelperfect#319)] works! There is one edge case where it does not work: Pick role, press back arrow and pick same role. Role is then denied.

**Comment by jessetilro** [comment]
Please see server PR for my response.

**Comment by dalderliesten** [comment]
Looks good to me.

**Pull request merged**